enumerables.rb

```diff
@@ -1,5 +1,107 @@
-  module Enumerable
+module Enumerable
     def my_each
-      # your code here
+      self.length.times do |x|
+        yield(self[x])
+      end
     end
-  end
\ No newline at end of file
+
+
+    def my_each_with_index
+      index = 0
+      self.length.times do |x|
+        yield(index)
+        index += 1
+      end
+    end
+
+
+    def my_select
+      arr = Array.new
+      self.length.times do |x|
+        if yield(self[x])
+          arr.push(x)
+        end
+      end
+      arr
+    end
+    
+
+    def my_all?
+      self.length.times do |x|
+        if !yield(self[x])
+          return false
+        else 
+          return true
+        end
+      end
+    end
+
+
+    def my_any?
+      self.length.times do |x|
+        if yield(self[x]) != false
+          return true
+        end
+      end
+      false
+    end
+
+
+    def my_none?
+      self.length.times do |x|
+        if yield(self[x]) 
+          return false
+        end
+      end
+      true
+    end
+
+
+    def my_count
```

Comment:
`my_count` does not return the proper input and it also should accept a default parameter